### PR TITLE
Search for chromium as CHROME_EXECUTABLE on Linux

### PR DESCRIPTION
**Pull request by xrmx**
Permit to use the distro provided chromium binary on linux in devicelab.

I don't know how to test this.

Refs #77900

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/devicelab/lib/framework/browser.dart [hunk 1]
```diff
@@ -218,13 +218,13 @@ String _findSystemChromeExecutable() {
 
   if (io.Platform.isLinux) {
     final io.ProcessResult which =
-        io.Process.runSync('which', <String>['google-chrome']);
+        io.Process.runSync('which', <String>['google-chrome', 'chromium']);
 
     if (which.exitCode != 0) {
```
Comment (xrmx) [inline dev/devicelab/lib/framework/browser.dart:223]: which returns 1 if not everyone command is found :facepalm:

**Comment by jonahwilliams** [comment]
Was the intention here to add support for chromium to the flutter_tool? If so this isn't the place to do it, devicelab is our test infrastructure.

**Comment by jonahwilliams** [comment]
(If so that is done here: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/web/chrome.dart#L28 )

**Comment by xrmx** [comment]
@jonahwilliams Indeed! Thanks for the hint.

**Comment by xrmx** [comment]
Moved to draft, at least this is failing:

```
30:49 +2824 ~39 -41: test/general.shard/web/devices_test.dart: Chrome invokes version command on non-Windows platforms [E]                                                                                        
  Expected: ['google-chrome', '--version']
    Actual: ['chromium', '--version']
     Which: at location [0] is 'chromium' instead of 'google-chrome'
```

**Comment by jonahwilliams** [comment]
I think you need to take a step back here. The existing code you are modifying is just written to look for executables based on a PATH. Adding in os calls to which directly is going to change the behavior a lot, and there might be better ways to handle this on linux

**Comment by xrmx** [comment]
Any idea on a better approach? I can't think of anything different than a runtime check.

**Comment by jonahwilliams** [comment]
I think a runtime check is a reasonable way to handle it, but the code needs to be adopted to support that.

**Comment by Hixie** [comment]
I'm going to close this PR since it's not ready for review, but please don't hesitate to create a new PR if you get back to it!

**Pull request closed without merging**